src/main/scala/com/gu/contentapi/utils/PredicateUtils.scala

```diff
@@ -0,0 +1,20 @@
+package com.gu.contentapi.utils
+
+object PredicateUtils {
+
+  implicit class RichPredicate[A](val f: Function1[A, Boolean]) extends AnyVal {
+    def apply(v: A) = f(v)
+
+    def &&(g: Function1[A, Boolean]): Function1[A, Boolean] = {
```

Comment:
It is used here
```scala
val downloadsLogs = allAcastLogs.filter(AcastLog.onlyDownloads && AcastLog.onlySuccessfulReponses)
```

where the `&&` operator is overloaded. I would personally get rid of the whole `RichPredicate` thing and rewrite that line with a good old

```scala
val x = allAcastLogs filter AcastLog.onlyDownloads filter AcastLog.onlySuccessfulReponses
```

because a) we're not using 3 of the 4 operators defined and b) I don't have to lookup what `filter` does so the code feels unnecessary complicated here. My 2 cents :)